cmd/prometheus/main.go

```diff
@@ -434,6 +435,16 @@ func main() {
 		os.Exit(1)
 	}
 
+	if agentMode && len(serverOnlyFlags) > 0 {
+		fmt.Fprintf(os.Stderr, "The following flag(s) can not be used in agent mode: %q", serverOnlyFlags)
+		os.Exit(3)
+	}
+
+	if !agentMode && len(agentOnlyFlags) > 0 {
+		fmt.Fprintf(os.Stderr, "The following flag(s) can only be used in agent mode: %q", agentOnlyFlags)
+		os.Exit(3)
+	}
```

Comment:
Ah, thanks for fixing this. I think this got lost at some point during fixing merge conflicts.